projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -145,14 +157,69 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     this.domMutationObserver.disconnect();
   }
 
+  public get isCodeEmpty(): boolean {
+    return isEmpty(this.code);
+  }
+
   public isLineHighlighted(lineNum: number): boolean {
     return (
       !isEmpty(this.highlightText) && this.codeLines[lineNum].toLowerCase().includes(this.highlightText.toLowerCase())
     );
   }
 
-  public onSearch(searchText: string): void {
-    this.searchText = searchText;
+  public onSearch(searchString: string): void {
+    // For case-insensitive search
+    const searchText = searchString.toLowerCase();
+    const codeTexts = this.codeTexts.map(text => text.toLowerCase());
+
+    // Remove existing child background elements
+    this.codeLineBackgroundElements.forEach(
+      codeLineBackgroundElement => (codeLineBackgroundElement.nativeElement.innerHTML = '')
+    );
+
+    const searchLen: number = searchText.length;
+    if (searchLen === 0) {
+      return;
+    }
+
+    const searchedPositions: Position[][] = Array.from(Array(codeTexts.length), () => []);
+
+    // Get all searched text positions
+    codeTexts.forEach((codeText, index) => {
+      for (let i = 0; i <= codeText.length - searchLen; i++) {
```

Comment:
nit: with this approach of finding matching indices, since you don't need to keep searching for every index, you can clean this up by doing a while loop on `searchedIndex > 1`.

That way, you can avoid the else - break.

Reply:
Added in the https://github.com/hypertrace/hypertrace-ui/pull/1574